lib/server/swarm.js

```diff
@@ -1,30 +1,38 @@
 module.exports = Swarm
 
 var debug = require('debug')('bittorrent-tracker')
+var LRU = require('lru')
 var randomIterate = require('random-iterate')
 
 // Regard this as the default implementation of an interface that you
 // need to support when overriding Server.createSwarm() and Server.getSwarm()
 function Swarm (infoHash, server) {
-  this.peers = {}
+  this.peers = new LRU({
+    max: server.peersCacheLength || 1000,
+    maxAge: server.peersCacheTtl || 900 // 900s = 15 minutes
+  })
   this.complete = 0
   this.incomplete = 0
 }
 
 Swarm.prototype.announce = function (params, cb) {
   var self = this
   var id = params.type === 'ws' ? params.peer_id : params.addr
-  var peer = self.peers[id]
+  // Mark the source peer as recently used in cache
+  var peer = self.peers.get(id)
 
-  if (params.event === 'started') {
+  // Get the peer back in swarm if missing
```

Comment:
This is already handled inside of `_onAnnounceCompleted` and `_onAnnounceUpdate`. No need to add back the peer here.